modules/sdp/SdpSimulcast.ts

```diff
@@ -0,0 +1,307 @@
+        });
+    
+        return mLine;
+    }
+
+    /**
+     * Returns a random number to be used for the SSRC.
+     *
+     * @returns
+     */
+    _generateSsrc() : number {
+        const min = 0, max = 0xffffffff;
+
+        return Math.floor(Math.random() * (max - min)) + min;
+    }
+
+    /**
+     * Returns the requested attribute value for a SSRC from a given media description.
+     *
+     * @param mLine
+     * @param ssrc
+     * @param attributeName
+     * @returns
+     */
+    _getSsrcAttribute(mLine: transform.MediaDescription, ssrc: number, attributeName: string) : string {
+        return mLine.ssrcs
+            .filter(ssrcInfo => parseInt(ssrcInfo.id?.toString()) === ssrc)
+            .filter(ssrcInfo => ssrcInfo.attribute === attributeName)
+            .map(ssrcInfo => ssrcInfo.value)[0];
+    }
+
+    /**
+     * Returns an array of all the primary SSRCs in the SIM group for a given media description.
+     *
+     * @param mLine
+     * @returns
+     */
+    _parseSimLayers(mLine: transform.MediaDescription) : Array<number> {
+        const simGroup = mLine.ssrcGroups?.find(group => group.semantics === 'SIM');
+
+        if (simGroup) {
+            return simGroup.ssrcs.split(' ').map(ssrc => parseInt(ssrc));
+        }
+
+        return [ parseInt(mLine.ssrcs?.[0]?.id?.toString()) ];
+    }
+
+    /**
+     * Munges the given media description to enable simulcast for the video media sections that are in either have
+     * SENDRECV or SENDONLY as the media direction thereby ignoring all the RECVONLY transceivers created for remote
+     * endpoints.
+     * NOTE: This needs to be called only when simulcast is enabled.
+     *
+     * @param description
+     * @returns
+     */
+    mungeLocalDescription(description: Description) : Description {
+        if (!(description && description?.sdp !== '')) {
+            return description;
+        }
+        const session = transform.parse(description.sdp);
+
+        for (let media of session.media) {
+            // Ignore recvonly and inactive transceivers created for remote sources.
+            if (media.direction === MediaDirection.RECVONLY || media.direction === MediaDirection.INACTIVE) {
+                continue;
+            }
+
+            // Ignore audio m-lines.
+            if (media.type !== MediaType.VIDEO) {
+                continue;
+            }
+            const mid = media.mid;
+            const numSsrcs = new Set(media.ssrcs?.map(ssrcInfo => ssrcInfo.id));
+            const numGroups = media.ssrcGroups?.length ?? 0;
+            let primarySsrc: number;
+
+            // Do not munge if the description has no ssrcs or if simulcast is already enabled.
+            if (numSsrcs.size === 0 || numSsrcs.size > 2 || (numSsrcs.size === 2 && numGroups === 0)) {
+                continue;
+            }
+            if (numSsrcs.size === 1) {
+                primarySsrc = parseInt((media.ssrcs[0]?.id).toString());
```

Comment:
again doing toString to parse it back, what will happen when `media.ssrcs[0]?.id` is `undefined`? Will it convert to string and parse as `NaN`?

Reply:
same reason as above, code copied from npm module, will change it.